packages/middleware-bucket-endpoint/src/bucketEndpointMiddleware.ts

```diff
@@ -0,0 +1,70 @@
+import { bucketHostname } from "./bucketHostname";
+import { BucketEndpointResolvedConfig } from "./configurations";
+import {
+  BuildHandler,
+  BuildHandlerArguments,
+  BuildHandlerOptions,
+  BuildHandlerOutput,
+  BuildMiddleware,
+  MetadataBearer,
+  Pluggable
+} from "@aws-sdk/types";
+import { HttpRequest } from "@aws-sdk/protocol-http";
+
+export function bucketEndpointMiddleware(
+  options: BucketEndpointResolvedConfig
+): BuildMiddleware<any, any> {
+  return <Output extends MetadataBearer>(
+    next: BuildHandler<any, Output>
+  ): BuildHandler<any, Output> => async (
+    args: BuildHandlerArguments<any>
+  ): Promise<BuildHandlerOutput<Output>> => {
+    const { Bucket: bucketName } = args.input;
+    let replaceBucketInPath =
+      options.preformedBucketEndpoint || options.bucketEndpoint;
+    let request = args.request;
+    if (HttpRequest.isInstance(request)) {
+      if (options.bucketEndpoint) {
+        request.hostname = bucketName;
+      } else if (!options.preformedBucketEndpoint) {
+        const { hostname, bucketEndpoint } = bucketHostname({
+          bucketName,
+          baseHostname: request.hostname,
+          accelerateEndpoint: options.useAccelerateEndpoint,
+          dualstackEndpoint: options.useDualstackEndpoint,
+          pathStyleEndpoint: options.forcePathStyle,
+          sslCompatible: request.protocol === "https:"
```

Comment:
This key should be `tlsCompatible` as we use `tls` everywhere now

Reply:
changed to ``tlsCompatible``